docs/guides/firmware-upgrade/readme.md

```diff
@@ -0,0 +1,100 @@
+---
+authors: Tony Floeder <anthony.floeder@hpe.com>
+categories: setup
+---
+
+# Firmware Upgrade Procedures
+
+This guide presents the firmware upgrade procedures to upgrade firmware from the Rabbit processor using tools present in the operating system.
+
+## NVMe Drive Firmware Upgrade
+
+In order to upgrade the firmware on NVMe drives attached to Rabbit, the `switchtec` and `switchtec-nvme` executables must be installed. All firmware downloads to drives are sent to the physical device of the drive which is accessible only using the `switchtec-nvme` executable.
+
+### Batch Method
+
+#### Download and Commit New Firmware
+
+The [nvme.sh](https://github.com/NearNodeFlash/nnf-ec/blob/master/tools/nvme.sh) helper script applies the same command to each physical device ID in the system. It provides a convenient way to upgrade the firmware on all drives in the system. Please see [fw-download](https://www.mankier.com/1/nvme-fw-download) and [fw-commit](https://www.mankier.com/1/nvme-fw-commit) for details about the individual commands.
+
+```bash
+# Download firmware to all drives
+./nvme.sh cmd fw-download --fw=</path/to/nvme.fw>
+
+# Commit the new firmware
+# action=3: The image is requested to be activated immediately
+./nvme.sh cmd fw-commit --action=3
+```
+
+#### Rebind the PCIe Connections
+
+In order to use the drives at this point, they must be unbound and bound to the PCIe fabric to reset device connections. The [bind.sh](https://github.com/NearNodeFlash/nnf-ec/blob/master/tools/bind.sh) helper script performs these two actions. Its use is illustrated below.
+
+```bash
+# Unbind all drives from the Rabbit to disconnect the PCIe connection to the drives
+./bind.sh unbind
+
+# Bind all drives to the Rabbit to reconnect the PCIe bus
+./bind.sh bind
+
+# At this point, your drives should be running the new firmware.
+# Verify the firmware...
+./nvme.sh cmd id-ctrl | grep -E "^fr "
+```
+
+### Individual Drive Method
+
+#### Determine Physical Device ID
+
+The first step is to determine a drive's physical device ID. The following code fragment demonstrates one way to list the physcial device ids of all the NVMe drives in the system.
```

Comment:
I forgot the 'F' when I told you what it stands for. If we spell it out it might make more sense what the bash commands are doing.

So maybe
> ... to determine a drive's unique Physical Device Fabric Identifier (PDFID).